experiment/selinux/src/xattrs/xattr.rs

```diff
@@ -0,0 +1,119 @@
+// set_xattr sets extended attributes on a file specified by its path.
+pub fn set_xattr(fpath: &Path, attr: &str, data: &[u8]) -> Result<(), XattrError> {
+    match rfs::setxattr(fpath, attr, data, rfs::XattrFlags::CREATE) {
+        Ok(_) => Ok(()),
+        Err(e) => {
+            let errno = e.raw_os_error();
+            if errno == libc::EINTR {
+                return Err(XattrError::EINTR(errno));
+            }
+            Err(XattrError::SetXattr(e.to_string()))
+        }
+    }
+}
+
+// function similar with lsetxattr in golang.org/x/sys/unix repo.
+// lset_xattr sets extended attributes on a symbolic link.
+pub fn lset_xattr(fpath: &Path, attr: &str, data: &[u8]) -> Result<(), XattrError> {
+    match rfs::lsetxattr(fpath, attr, data, rfs::XattrFlags::CREATE) {
+        Ok(_) => Ok(()),
+        Err(e) => {
+            let errno = e.raw_os_error();
+            if errno == libc::EINTR {
+                return Err(XattrError::EINTR(errno));
+            }
+            Err(XattrError::LSetXattr(e.to_string()))
+        }
+    }
+}
+
+// function similar with getattr in go-selinux repo.
+// get_xattr returns the value of an extended attribute attr set for path.
+pub fn get_xattr(fpath: &Path, attr: &str) -> Result<String, XattrError> {
+    // SELinux label is not so big, so we allocate 1024 bytes for the buffer.
+    let mut buf_size = 1024;
+    let mut buf = vec![0u8; buf_size];
+
+    loop {
+        match rfs::getxattr(fpath, attr, &mut buf) {
+            Ok(size) => {
+                if size == buf_size {
+                    buf_size *= 2;
+                    buf.resize(buf_size, 0);
+                    continue;
+                }
+                let mut value = String::from_utf8_lossy(&buf[..size]).into_owned();
+                if value.ends_with('\x00') {
+                    value.pop();
+                }
+                return Ok(value);
+            }
+            Err(e) => return Err(XattrError::GetXattr(e.to_string())),
+        }
+    }
+}
+
+// function similar with lgetxattr in go-selinux repo.
+// lget_xattr returns the value of an extended attribute attr set for path.
+pub fn lget_xattr(fpath: &Path, attr: &str) -> Result<String, XattrError> {
+    // SELinux label is not so big, so we allocate 1024 bytes for the buffer.
+    let mut buf_size = 1024;
+    let mut buf = vec![0u8; buf_size];
+
+    loop {
+        match rfs::lgetxattr(fpath, attr, &mut buf) {
+            Ok(size) => {
+                if size == buf_size {
+                    buf_size *= 2;
+                    buf.resize(buf_size, 0);
+                    continue;
+                }
+                let mut value = String::from_utf8_lossy(&buf[..size]).into_owned();
+                if value.ends_with('\x00') {
+                    value.pop();
+                }
+                return Ok(value);
+            }
+            Err(e) => return Err(XattrError::LGetXattr(e.to_string())),
+        }
+    }
+}
```

Comment:
What do you think about implementing it as a trait for `Path` or `File`?
Like that:

```rust
pub trait PathBufExt {
    lget_xattr(&self, attr: &str) -> Result<String, XattrError>
...
}

impl PathBufExt for Path {
	fn lget_xattr(&self, attr: &str) -> Result<String, XattrError> {
		...
	}
	...
}
```
https://github.com/containers/youki/blob/a2390311b44f9beacd7b801bea0078daf8bbc1b4/crates/libcontainer/src/utils.rs#L35-L42

Reply:
WDYT? @YJDoc2 @Gekko0114

Reply:
Yeah, sure. I think this change will make the code easier to handle.

Reply:
Fixed it